src/main/java/com/aws/greengrass/secretmanager/AWSSecretClient.java

```diff
@@ -6,38 +6,40 @@
 package com.aws.greengrass.secretmanager;
 
 import com.aws.greengrass.deployment.DeviceConfiguration;
+import com.aws.greengrass.logging.api.Logger;
+import com.aws.greengrass.logging.impl.LogManager;
 import com.aws.greengrass.secretmanager.exception.SecretManagerException;
 import com.aws.greengrass.tes.LazyCredentialProvider;
-import com.aws.greengrass.util.BaseRetryableAccessor;
 import com.aws.greengrass.util.Coerce;
-import com.aws.greengrass.util.CrashableSupplier;
 import com.aws.greengrass.util.ProxyUtils;
+import com.aws.greengrass.util.RetryUtils;
 import com.aws.greengrass.util.Utils;
 import software.amazon.awssdk.regions.Region;
 import software.amazon.awssdk.services.secretsmanager.SecretsManagerClient;
-import software.amazon.awssdk.services.secretsmanager.model.DecryptionFailureException;
 import software.amazon.awssdk.services.secretsmanager.model.GetSecretValueRequest;
 import software.amazon.awssdk.services.secretsmanager.model.GetSecretValueResponse;
 import software.amazon.awssdk.services.secretsmanager.model.InternalServiceErrorException;
-import software.amazon.awssdk.services.secretsmanager.model.InvalidParameterException;
-import software.amazon.awssdk.services.secretsmanager.model.InvalidRequestException;
 import software.amazon.awssdk.services.secretsmanager.model.ResourceNotFoundException;
 
 import java.io.IOException;
-import java.util.Collections;
-import java.util.HashSet;
+import java.time.Duration;
+import java.util.Arrays;
 import javax.inject.Inject;
 
 public class AWSSecretClient {
 
     private final SecretsManagerClient secretsManagerClient;
-    private static final int RETRY_COUNT = 3;
-    private static final int BACKOFF_MILLIS = 200;
+    private final Logger logger = LogManager.getLogger(AWSSecretClient.class);
+    private final RetryUtils.RetryConfig retryConfig =
+            RetryUtils.RetryConfig.builder().maxAttempt(3).maxRetryInterval(Duration.ofMillis(200)).retryableExceptions(
+                    Arrays.asList(IOException.class, InternalServiceErrorException.class,
+                            ResourceNotFoundException.class)).build();
```

Comment:
Does the ResourceNotFoundException come from the cloud? If it does, are we expecting that the resource will be found on retry?

Reply:
Yes, it comes from cloud. My idea is to retry thrice before giving it up.